packages/light-client-extension-helpers/src/smoldot/tasks.ts

```diff
@@ -0,0 +1,68 @@
+import { Client, AddChainError } from "."
+import { Effect, Fiber, Runtime, Schedule } from "effect"
+
+export const DEFAULT_SUPERVISE_REPEAT_SCHEDULE = Schedule.spaced("1 second")
+export const DEFAULT_SUPERVISE_RETRY_SCHEDULE = Schedule.spaced("1 second")
+
+export type SuperviseOptions = {
+  repeatSchedule: Schedule.Schedule<number>
+  retrySchedule: Schedule.Schedule<number>
+}
+
+/**
+ * Supervises a smoldot client by periodically invoking `addChain` with an
+ * empty chain spec.
+ *
+ * If `addChain` fails with anything other than a `AddChainError`, the client
+ * will be restarted.
+ *
+ * @param client - The smoldot client instance to be supervised.
+ * @param options - Options for customizing the supervision behavior.
+ * @param options.repeatSchedule - The frequency at which to invoke `addChain`.
+ * Defaults to {@link DEFAULT_SUPERVISE_REPEAT_SCHEDULE}.
+ * @param options.retrySchedule - The frequency at which to attempt restarting
+ * the client if needed. Defaults to {@link DEFAULT_SUPERVISE_RETRY_SCHEDULE}.
+ */
+export const supervise = async (
```

Comment:
Nit: this file is called `tasks.ts` but the main export is this supervise stuff; maybe call it `supervise.ts` or similar?

Reply:
Yeah thats possible but I dislike the Java-esque style where 1 file is 1 class or 1 function. If i treat it as a module then this is a task module and it has a supervise function and the module can be extended in the future to have more tasks.